File: ServiGo/backend/apps/usuarios/models/profesional.py
from django.db import models

from ..data.tipos_negocio import TipoNegocio
from .usuario import Usuario


class Profesional(models.Model):
    usuario = models.OneToOneField(
        Usuario,
        on_delete=models.CASCADE,
        related_name="profesional"
    )

    tipo_negocio = models.CharField(
        max_length=20,
        choices=TipoNegocio.choices,
        default=TipoNegocio.OTRO
    )

    nombre_negocio = models.CharField(max_length=150)
    descripcion = models.TextField(blank=True)
    telefono = models.CharField(max_length=20, blank=True)
    direccion = models.CharField(max_length=200, blank=True)
    imagen = models.URLField(blank=True)

    enlace_personalizado = models.SlugField(
        max_length=100,
        unique=True
    )

    activo = models.BooleanField(default=True)

    def __str__(self):
        return self.nombre_negocio

File: backend/src/models/Profesional.ts
import { Schema, model, Document, Types } from "mongoose";
import { TIPOS_NEGOCIO } from "../data/tiposNegocio";

export interface IProfesional extends Document {
  usuario: Types.ObjectId;
  tipoNegocio: string;
  nombreNegocio: string;
  descripcion?: string;
  telefono?: string;
  direccion?: string;
  imagen?: string;
  enlacePersonalizado: string;
  activo: boolean;
}

const profesionalSchema = new Schema<IProfesional>(
  {
    usuario: { type: Schema.Types.ObjectId, ref: "Usuario", required: true, unique: true },
    tipoNegocio: { type: String, enum: TIPOS_NEGOCIO, default: "OTRO" },
    nombreNegocio: { type: String, required: true, trim: true },
    descripcion: { type: String, default: "" },
    telefono: { type: String, default: "" },
    direccion: { type: String, default: "" },
    imagen: { type: String, default: "" },
    enlacePersonalizado: { type: String, required: true, unique: true, trim: true },
    activo: { type: Boolean, default: true },
  },
  { timestamps: true }
);

export const Profesional = model<IProfesional>("Profesional", profesionalSchema);
